wicket-util/src/main/java/org/apache/wicket/util/crypt/SunJceCrypt.java

```diff
@@ -153,4 +194,18 @@ protected KeySpec createKeySpec()
 	{
 		return new PBEKeySpec(getKey().toCharArray());
 	}
+
+	/**
+	 * Create a random salt to be used for this crypt. 
+	 * 
+	 * @return salt, always 8 bytes long
+	 */
+	public static byte[] randomSalt()
+	{
+		// must be 8 bytes - for anything else PBES1Core throws
+		// InvalidAlgorithmParameterException: Salt must be 8 bytes long  
+		byte[] salt = new byte[8];
+		new Random().nextBytes(salt);
```

Comment:
Should we use CipherUtils.randomByteArray(8) here ? It would use SecureRandom

Reply:
Better yet, remove this method entirely. Usage of this method should be replaced by ISecureRandomSupplier.getRandomBytes(length). The optimal length of the salt depends on the algorithm used. Some might require 8 bytes, but for example PBKDF2WithHmacSHA512 can use much longer salts and it's recommended to use a salt that's equal in length to your derived key.

Reply:
Since this is a legacy class I would leave it as it is as it can not be moved to wicket-core